### ISSUE TITLE
To Do

### ISSUE THREAD

**Issue by bnortz** [dragonrollers/thoughtbubble-app#16]
- [x] **Fix adapters to hold more than just question text (question text, answer text, question ID) **
- [x] Question, answer, and question ID is not yet persistent in the backend side. I hardcoded one question but it would eventually be needed to be fixed.
- [x] **fix adapter for friends list to hold friend name and ID** (`Discover` and `Answer` tabs)
    - Idea: If can't hold info that it's not displaying, then keep ID in a separate array that's updated at the same time as the adapter array and use indices to get ID information (like currently doing with friends list)

- [x] Create activity for user to see their own asked questions

- [x] Connect "their own asked questions" with database (needs special adapter) (`Discover` tab)

- [x] Connect detailed question page to database (needs adapter fix)

- [x] Implement answer question (needs adapter fix) (`Answer` tab)

- [x] Change Discover to show only answered questions by friends (also make it so it changes an array and notifies adapter instead of just updating adapter) (`Discover` tab)
    - Idea: Bonnie's idea to help with this: add a new section of the db that's called "answered questions" that has the IDs of all answered questions, and a boolean "changed" so that any listener on that portion can get updated when the question is updated in the main "questions" part of the database
    - Similarly, each user's incoming and outgoing questions should probably have a "changed" boolean

- [x] Connect profile to database (`User` tab)

- [x] Get profile photos and names from Facebook account.

- [x] Save changes whenever user changes their name, photos, or interests. (`User` tab)

- [x] Save names of the boards and their contents to Firebase. (`User` tab)

- [x] Search for new friends (hamburger menu)

- [x] Would be nice if we could figure out how to handle the issue of people sending empty questions (right now it's OK for them to send blank questions)

- [x] Adding an interest does not update the user profile interests

- [x] Make fragment layout for saving to a board (`Discover` tab)

- [x] Make boards smaller and let second board on right show more so users know they can swipe (`User` tab)

- [x] Split `Discover` tab into two views: 1) your asked questions 2) other people's questions

- [x] Search for boards in list mode (`User` tab)

- [x] Change AskSelect to dynamically load friends' names based on given user ids. Currently their names are also stored and won't be updated when that friend edits their name.

- [x] Change select friends to show profile pictures, (and "ask me about"s???)

**Comment by jennykim1016** [comment]
In addition to these, for firebase,

- [ ] We need to get profile photos and name from Facebook account (which is the one in the bottom). In addition to that, we would also need to save the change whenever user changes their name, photos, or interests.

- [ ] we need to save the names of the boards and their contents to firebase.

**Comment by bnortz** [comment]
@jennykim1016 The hamburger (which currently just shows up on the discover page) does have a button for signing out, and once a user signs out they can't do anything with the app until they sign in again

**Comment by jennykim1016** [comment]
Aha, perfect! I will look at the master branch!

**Comment by potsui** [comment]
@bnortz @jennykim1016 @gracehong Comprehensive list of TODOs, consolidating with all the other open issues we had

**Comment by jennykim1016** [comment]
I will start working on the first checkbox -- "Fix adapters to hold more than just question text (question text, answer text, question ID), fix adapter for friends list to hold friend name and ID".

**Comment by potsui** [comment]
Working on pulling user from database to populate profile page

**Comment by potsui** [comment]
You can now delete an interest/topic in the database but not sure how to pass in the `id` to the `DialogFragmentProfile`'s `onClickListener`, if anyone can just change that string the delete functionality will be all done.

**Comment by jennykim1016** [comment]
Still not sure about "fix adapter for friends list to hold friend name and ID (Discover and Answer tabs)"; are we done with this? @bnortz @potsui @gracehong

**Comment by potsui** [comment]
Checked off most items, carrying over incomplete ones to their own issues to assign to people

**Comment by potsui** [comment]
@jennykim1016 whoops didn't see your comment until after I closed and refreshed the page --- this is not done yet and has its own issue, because we need to restructure how we're currently storing friends. Basically we've been loading their names not IDs which has caused a discrepancy when a user changes their name. @bnortz and I already discussed how we're going to do that. CC: @gracehong 

See issue #35
